state/src/main.rs

```diff
@@ -15,21 +15,22 @@ extern crate actix;
 extern crate actix_web;
 extern crate env_logger;
 
-use std::cell::Cell;
+use std::sync::Arc;
+use std::sync::Mutex;
 
 use actix_web::{middleware, server, App, HttpRequest, HttpResponse};
 
 /// Application state
 struct AppState {
-    counter: Cell<usize>,
+    counter: Arc<Mutex<usize>>,
```

Comment:
To avoid further confusion, let's make sure that people understand why state here should be `Arc`

Please add comment that in order for counter to be shared among all HTTP workers, we wrap it into `Arc<Mutext<T>>`

Reply:
Ah you're right, I didn't check the whole file.
Then yes, leave it  as it is